src/rest_vol_dataset.c

```diff
@@ -389,15 +389,15 @@ RV_dataset_open(void *obj, const H5VL_loc_params_t *loc_params, const char *name
  *              March, 2017
  */
 herr_t
-RV_dataset_read(void *obj, hid_t mem_type_id, hid_t mem_space_id,
-    hid_t file_space_id, hid_t dxpl_id, void *buf, void **req)
+RV_dataset_read(size_t count, void *dset[], hid_t mem_type_id[], hid_t mem_space_id[],
+                   hid_t file_space_id[], hid_t dxpl_id, void *buf[], void **req)
 {
     H5S_sel_type  sel_type = H5S_SEL_ALL;
-    RV_object_t  *dataset = (RV_object_t *) obj;
+    RV_object_t  *dataset = (RV_object_t *) dset[0];
     H5T_class_t   dtype_class;
     hssize_t      mem_select_npoints, file_select_npoints;
     hbool_t       is_transfer_binary = FALSE;
-    htri_t        is_variable_str;
+    hbool_t        is_variable_str;
```

Comment:
Same here for the `htri_t` -> `hbool_t` issue. Apologies if I wasn't very clear on that; `htri_t` is still used in different places in the library, we just changed the two arguments for the "link exists" and "attribute exists" callbacks from `htri_t` -> `hbool_t`.